addon/components/connect.js

```diff
@@ -5,36 +5,46 @@ const {
   computed,
   defineProperty,
   inject: { service },
-  isEmpty,
   run
 } = Ember;
 
 export default (stateToComputed, dispatchToActions=() => ({})) => {
-
-  if (!stateToComputed) {
-    stateToComputed = () => ({});
-  }
-
   return Component => {
-
     return Component.extend({
-
       redux: service(),
 
       init() {
         const redux = this.get('redux');
 
-        let props = stateToComputed.call(this, redux.getState(), this.getAttrs());
+        if (stateToComputed) {
+          const getProps = () => stateToComputed.call(this, redux.getState(), this.getAttrs());
+          let props = getProps();
+
+          Object.keys(props).forEach(name => {
+            defineProperty(this, name, computed(() =>
+              props[name]
+            ).property().readOnly());
+          });
+
+          this._handleChange = () => {
+            const newProps = getProps();
+            if (props === newProps) return;
```

Comment:
This is the only line I don't see adding value visibly (if I console log when this condition is true I never see anything showing this is useful). That said, I do understand it conceptually (value based equality with immutability). Are you able to see this saving cpu cycles down the road on big apps?